docs/user/openstack/troubleshooting.md

```diff
@@ -0,0 +1,68 @@
+# OpenShift 4 installer on OpenStack troubleshooting
+
+Support for launching clusters on OpenStack is **experimental**.
+
+Unfortunately, there will always be some cases where OpenShift fails to install properly. In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
+
+This document discusses some troubleshooting options for OpenStack based
+deployments.
+
+## View instances logs
+
+OpenStack CLI tools should be installed, then:
+
+`openstack console log show <instance>`
+
+## ssh access to the instances
+
+By default, the only exposed instance is the service VM, but ssh access is not
+allowed (nor the ssh key injected), so in case you want to access the hosts, it
+is required to create a floating IP and attach it to some host (master-0 in
+this example)
+
+### Create security group to allow ssh access
+
+```
+export INSTANCE=$(openstack server list -f value -c Name | grep master-0)
+openstack security group create ssh
+openstack security group rule create \
+  --ingress \
+  --protocol tcp \
+  --dst-port 22 \
+  ssh
+openstack server add security group ${INSTANCE} ssh
+```
+
+Optionally, allow ICMP traffic (to ping the instance):
+
+```
+openstack security group rule create \
+  --ingress \
+  --protocol icmp \
+  ssh
+```
+
+### Create and attach the floating IP
+
+```
+# This assumes there are only two networks, the openshift one and the public one
+export PUBLIC_NETWORK=$(openstack network list -f value -c Name | grep -v openshift)
```

Comment:
This doesn't look right.  Wouldn't `openshift` be the private network?

Reply:
Correct. Prolly just s/openshift/<public-network>/

Reply:
Exactly. The command will get the list of networks and will select (grep -v) the one that is not openshift